src/Raven.Server/Rachis/Elector.cs

```diff
@@ -30,249 +34,271 @@ public override string ToString()
             return $"Elector {_engine.Tag} for {_connection.Source}";
         }
 
-        public void HandleVoteRequest()
+        private void HandleVoteRequest()
         {
             try
             {
-                while (true)
+                try
+                {
+                    Thread.CurrentThread.Priority = ThreadPriority.AboveNormal;
+                }
+                catch (Exception e)
                 {
-                    using (_engine.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                    if (_engine.Log.IsInfoEnabled)
                     {
-                        var rv = _connection.Read<RequestVote>(context);
-
-                        if (_engine.Log.IsInfoEnabled)
-                        {
-                            var election = rv.IsTrialElection ? "Trial" : "Real";
-                            _engine.Log.Info($"Received ({election}) 'RequestVote' from {rv.Source}: Election is {rv.ElectionResult} in term {rv.Term} while our current term is {_engine.CurrentTerm}, " +
-                                             $"Forced election is {rv.IsForcedElection}. (Sent from:{rv.SendingThread})");
-                        }
+                        _engine.Log.Info("Elector was unable to set the thread priority, will continue with the same priority", e);
+                    }
+                }
 
-                        //We are getting a request to vote for our known leader
-                        if (_engine.LeaderTag == rv.Source)
+                using (this)
```

Comment:
You still need to dispose elector externally if you have this?

Reply:
This is a background task. Same situation with Follower.